jetty-core/jetty-tests/jetty-test-client-transports/src/test/java/org/eclipse/jetty/test/client/transport/HttpClientStreamTest.java

```diff
@@ -1065,6 +1059,99 @@ public void close() throws IOException
         assertTrue(closeLatch.await(5, TimeUnit.SECONDS));
     }
 
+    @ParameterizedTest
+    @MethodSource("transports")
+    public void testUploadWithPendingReadConcurrentServerCloseClosesStream(Transport transport) throws Exception
+    {
+        CountDownLatch serverDemandLatch = new CountDownLatch(1);
```

Comment:
What's this for?
Should you not count it down in the server-side demand callback?